pkg/scheduler/framework/runtime/framework.go

```diff
@@ -1421,29 +1421,67 @@ func (f *frameworkImpl) RunPreBindPlugins(ctx context.Context, state fwk.CycleSt
 		logger = klog.LoggerWithName(logger, "PreBind")
 		logger = klog.LoggerWithValues(logger, "node", klog.ObjectRef{Name: nodeName})
 	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	statusCh := parallelize.NewResultChannel[*fwk.Status]()
+	skippedPlugins := state.GetSkipPreBindPlugins()
+	parallelPlugins := state.GetParallelPreBindPlugins()
+
+	pluginGroups := f.getPreBindPluginGroups(state, skippedPlugins, parallelPlugins)
+	for _, plugins := range pluginGroups {
+		f.Parallelizer().Until(ctx, len(plugins), func(index int) {
+			pl := plugins[index]
+			ctx := ctx
+			if verboseLogs {
+				logger := klog.LoggerWithName(logger, pl.Name())
+				ctx = klog.NewContext(ctx, logger)
+			}
+			plStatus := f.runPreBindPlugin(ctx, pl, state, pod, nodeName)
+			if !plStatus.IsSuccess() {
+				if plStatus.IsRejected() {
+					logger.V(4).Info("Pod rejected by PreBind plugin", "pod", klog.KObj(pod), "node", nodeName, "plugin", pl.Name(), "status", plStatus.Message())
+					plStatus.SetPlugin(pl.Name())
+					statusCh.SendWithCancel(plStatus, cancel)
+					return
+				}
+				err := plStatus.AsError()
+				logger.Error(err, "Plugin failed", "plugin", pl.Name(), "pod", klog.KObj(pod), "node", nodeName)
+				statusCh.SendWithCancel(fwk.AsStatus(fmt.Errorf("running PreBind plugin %q: %w", pl.Name(), err)), cancel)
+			}
+		}, metrics.PreBind)
+		if status := statusCh.Receive(); status != nil {
+			return status
+		}
+	}
+	return nil
+}
+
+// getPreBindPluginGroups chunks the plugins implementing the PreBind extension point
+// into groups of plugins that can be run in parallel.
+// The returned value consists of non-skipped plugins only.
+func (f *frameworkImpl) getPreBindPluginGroups(state fwk.CycleState, skippedPlugins, parallelPlugins sets.Set[string]) [][]fwk.PreBindPlugin {
+	var groups [][]fwk.PreBindPlugin
+	var group []fwk.PreBindPlugin
 	for _, pl := range f.preBindPlugins {
```

Comment:
nit: handle skipped plugins to simplify a bit?

```
for _, pl := range f.preBindPlugins {
  if skippedPlugins.Has(pl.Name()) {
    continue
  }

...

```

Reply:
That change would break the assumption we made in https://github.com/kubernetes/kubernetes/pull/135393#issuecomment-3605884594 - which is also exercised in the last two unit test cases in `framework_test.go`.